contracts/SnappAuction.sol

```diff
@@ -68,13 +70,39 @@ contract SnappAuction is SnappBase {
         auctions[auctionIndex].size++;
     }
 
+    function applyAuction(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 _newStateRoot,
+        bytes32 _orderHash,
+        uint128[MAX_TOKENS] memory prices,
+        uint128[2*AUCTION_BATCH_SIZE] memory volumes
+    )
+        public onlyOwner()
+    {   
+        require(slot <= auctionIndex, "Requested order slot does not exist");
+        require(slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0, "Must apply auction slots in order!");
+        require(auctions[slot].shaHash == _orderHash, "Order hash doesn't agree");
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");
+        require(block.number > auctions[slot].creationBlock + 20, "Requested order slot is still active");
+        require(stateRoots[stateIndex()] == _currStateRoot, "Incorrect state root");
+
+        stateRoots.push(_newStateRoot);        
+        auctions[slot].appliedAccountStateIndex = stateIndex();
+        
+        emit AuctionSolution(slot, prices, volumes);
+        emit StateTransition(TransitionType.Auction, stateIndex(), _newStateRoot, slot);
```

Comment:
What's the reasoning behind having two different events here? Isn't this update one atomic transaction that should invoke exactly one callback inside the event listener?

The way it's designed right now, you'd have partial data in two different callbacks.

Reply:
The reasoning here is to use the existing `StateTransition` event of type Auction. I mostly submitted this to get some feedback and idea about how we might do this differently (i.e. with a single event). It looks like we are going to have to scrap the StateTransition of type Auction...

Reply:
Done!